paths resolution is not working properly

### Describe the bug

`paths` resolution is not working for me.

I've provided paths config as

```json
{
 "paths": {
      "@moduleA/*": ["modules/01-moduleA/*"],
      "@moduleB/*": ["modules/02-moduleB/*"]
    }
}
```

When I'm trying to import using aliases, the compilation fails:

```
import { sum } from "@moduleA/utils";
import { multiply } from "@moduleB/utils";

console.log(sum(1, 2));
console.log(multiply(1, 2));
```

The reference to `@moduleA/utils` is being read as `@moduleA/utilsoduleA/utils`

Error:
```
ERROR in ./src/index.ts 1:0-49
Module not found: Error: Can't resolve '@moduleA/utilsoduleA/utils' in '/Users/vivek.bansal/Projects/vkbansal/swc-test/src'
resolve '@moduleA/utilsoduleA/utils' in '/Users/vivek.bansal/Projects/vkbansal/swc-test/src'
  Parsed request is a module
  using description file: /Users/vivek.bansal/Projects/vkbansal/swc-test/package.json (relative path: ./src)
    Field 'browser' doesn't contain a valid alias configuration
    resolve as module
      /Users/vivek.bansal/Projects/vkbansal/swc-test/src/node_modules doesn't exist or is not a directory
      looking for modules in /Users/vivek.bansal/Projects/vkbansal/swc-test/node_modules
        /Users/vivek.bansal/Projects/vkbansal/swc-test/node_modules/@moduleA/utilsoduleA doesn't exist
      /Users/vivek.bansal/Projects/vkbansal/node_modules doesn't exist or is not a directory
      /Users/vivek.bansal/Projects/node_modules doesn't exist or is not a directory
      /Users/vivek.bansal/node_modules doesn't exist or is not a directory
      /Users/node_modules doesn't exist or is not a directory
      /node_modules doesn't exist or is not a directory

ERROR in ./src/index.ts 2:0-54
Module not found: Error: Can't resolve '@moduleB/utilsoduleB/utils' in '/Users/vivek.bansal/Projects/vkbansal/swc-test/src'
resolve '@moduleB/utilsoduleB/utils' in '/Users/vivek.bansal/Projects/vkbansal/swc-test/src'
  Parsed request is a module
  using description file: /Users/vivek.bansal/Projects/vkbansal/swc-test/package.json (relative path: ./src)
    Field 'browser' doesn't contain a valid alias configuration
    resolve as module
      /Users/vivek.bansal/Projects/vkbansal/swc-test/src/node_modules doesn't exist or is not a directory
      looking for modules in /Users/vivek.bansal/Projects/vkbansal/swc-test/node_modules
        /Users/vivek.bansal/Projects/vkbansal/swc-test/node_modules/@moduleB/utilsoduleB doesn't exist
      /Users/vivek.bansal/Projects/vkbansal/node_modules doesn't exist or is not a directory
      /Users/vivek.bansal/Projects/node_modules doesn't exist or is not a directory
      /Users/vivek.bansal/node_modules doesn't exist or is not a directory
      /Users/node_modules doesn't exist or is not a directory
      /node_modules doesn't exist or is not a directory

webpack 5.68.0 compiled with 2 errors and 1 warning in 1372 ms
```

Repro Code: https://github.com/vkbansal/swc-test
Sandbox: https://codesandbox.io/s/github/vkbansal/swc-test/tree/main

### Input code

_No response_

### Config

```json
{
  "jsc": {
    "baseUrl": "./src",
    "target": "es2018",
    "externalHelpers": true,
    "parser": {
      "syntax": "typescript",
      "tsx": false,
      "decorators": false,
      "dynamicImport": true
    },
    "transform": {
      "react": {
        "runtime": "classic",
        "throwIfNamespace": true
      }
    },
    "paths": {
      "@moduleA/*": ["modules/01-moduleA/*"],
      "@moduleB/*": ["modules/03-moduleB/*"]
    }
  },
  "module": {
    "type": "es6",
    "strict": false,
    "strictMode": true,
    "lazy": false,
    "noInterop": false,
    "ignoreDynamic": true
  }
}
```


### Playground link

https://codesandbox.io/s/github/vkbansal/swc-test/tree/main

### Expected behavior

Project should compile without any errors

### Actual behavior

_No response_

### Version

1.2.136

### Additional context

_No response_

--- a/crates/swc/tests/fixture/paths/ext/001/output/index.js
+++ b/crates/swc/tests/fixture/paths/ext/001/output/index.js
@@ -1,2 +1,2 @@
-import styles from './styles/global.css';
+import styles from "./styles/global.css";
 console.log(styles);

--- a/crates/swc/tests/fixture/paths/ext/002/output/index.js
+++ b/crates/swc/tests/fixture/paths/ext/002/output/index.js
@@ -1,2 +1,2 @@
-import styles from './foo.ts/index';
+import styles from "./foo.ts/index";
 console.log(styles);

--- a/crates/swc/tests/fixture/paths/issue-3447/input/.swcrc
+++ b/crates/swc/tests/fixture/paths/issue-3447/input/.swcrc
@@ -1,0 +1,31 @@
+{
+  "jsc": {
+    "baseUrl": "./src",
+    "target": "es2018",
+    "externalHelpers": true,
+    "parser": {
+      "syntax": "typescript",
+      "tsx": false,
+      "decorators": false,
+      "dynamicImport": true
+    },
+    "transform": {
+      "react": {
+        "runtime": "classic",
+        "throwIfNamespace": true
+      }
+    },
+    "paths": {
+      "@moduleA/*": ["modules/01-moduleA/*"],
+      "@moduleB/*": ["modules/03-moduleB/*"]
+    }
+  },
+  "module": {
+    "type": "es6",
+    "strict": false,
+    "strictMode": true,
+    "lazy": false,
+    "noInterop": false,
+    "ignoreDynamic": true
+  }
+}

--- a/crates/swc/tests/fixture/paths/issue-3447/input/src/index.ts
+++ b/crates/swc/tests/fixture/paths/issue-3447/input/src/index.ts
@@ -1,0 +1,4 @@
+import { sum } from "@moduleA/utils";
+export { multiply } from "@moduleB/utils";
+
+console.log(sum(1, 2));

--- a/crates/swc/tests/fixture/paths/issue-3447/input/src/modules/01-moduleA/utils.ts
+++ b/crates/swc/tests/fixture/paths/issue-3447/input/src/modules/01-moduleA/utils.ts
@@ -1,0 +1,3 @@
+export function sum(a: number, b: number): number {
+  return a + b;
+}

--- a/crates/swc/tests/fixture/paths/issue-3447/input/src/modules/02-moduleB/utils.ts
+++ b/crates/swc/tests/fixture/paths/issue-3447/input/src/modules/02-moduleB/utils.ts
@@ -1,0 +1,3 @@
+export function multiply(a: number, b: number): number {
+  return a * b;
+}

--- a/crates/swc/tests/fixture/paths/issue-3447/output/src/index.ts
+++ b/crates/swc/tests/fixture/paths/issue-3447/output/src/index.ts
@@ -1,0 +1,3 @@
+import { sum } from "./modules/01-moduleA/utils";
+export { multiply } from "./modules/03-moduleB/utils";
+console.log(sum(1, 2));

--- a/crates/swc/tests/fixture/paths/issue-3447/output/src/modules/01-moduleA/utils.ts
+++ b/crates/swc/tests/fixture/paths/issue-3447/output/src/modules/01-moduleA/utils.ts
@@ -1,0 +1,3 @@
+export function sum(a, b) {
+    return a + b;
+}

--- a/crates/swc/tests/fixture/paths/issue-3447/output/src/modules/02-moduleB/utils.ts
+++ b/crates/swc/tests/fixture/paths/issue-3447/output/src/modules/02-moduleB/utils.ts
@@ -1,0 +1,3 @@
+export function multiply(a, b) {
+    return a * b;
+}

--- a/crates/swc_ecma_transforms_module/src/rewriter.rs
+++ b/crates/swc_ecma_transforms_module/src/rewriter.rs
@@ -53,6 +53,7 @@ where
             .with_context(|| format!("failed to resolve import `{}`", i.src.value))
             .unwrap();
 
+        i.src.kind = Default::default();
         i.src.value = src;
     }
 
@@ -64,6 +65,7 @@ where
                 .with_context(|| format!("failed to resolve import `{}`", src.value))
                 .unwrap();
 
+            src.kind = Default::default();
             src.value = new;
         }
     }

